src/api/Api.js

```diff
@@ -229,14 +229,27 @@ class Api {
             requestUrl = requestUrl.substr(0, requestUrl.indexOf('?'));
         }
 
+        // Encode existing query parameters, since tomcat does not accept unencoded brackets. Throw
+        // an error if they're already encoded to prevent double encoding.
+        if (query) {
+            const isEncoded = query !== decodeURIComponent(query);
```

Comment:
decodeURIComponent can throw an error does that need to be handled separately? E.g. `decodeURIComponent('test%5')`

Reply:
Ah yeah you're right, would probably be nicer to return a rejected promise, to keep the api consistent.

Reply:
I'm now rejecting with an error with the message 'Query parameters in URL are invalid' here. I found decodeURIComponent's 'URI malformed' a bit too unclear.